### PR TITLE
Improve developer experience for previewing docs in development

### PR DESCRIPTION
**Pull request by nholden**
## What type of PR is this?

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [X] Documentation Update

## Description

There is [no documentation on how to preview the docs in development](https://github.com/thepracticaldev/dev.to/pull/819#issuecomment-427081402). It turns out that it requires installing an npm package globally, navigating to a directory, and running a command. The `@next` version of that npm package, `gitdocs`, [is currently broken](https://github.com/timberio/gitdocs/issues/156#issuecomment-427611528), so even if someone knew all the steps to preview the docs, it's likely they would hit an error.

This change does a few things to help make previewing docs in development a less painful experience:

1. It adds `gitdocs` as a dev dependency so that we can control the version and ensure it works with our configs.
2. It adds a Yarn script so that serving the docs is a one liner for developers and easily configurable if we need to change the implementation in the future.
3. It adds documentation about how to preview the docs.

## Desktop Screenshot

screen shot 2018-10-06 at 4 16 53 pm

## Added to documentation?

- [X] docs.dev.to
- [ ] readme
- [ ] no documentation needed

### CODE REVIEW

**Comment by benhalpern** [comment]
@jessleenyc @maestromac can you give this a review?

File: docs/get-api-keys-dev-env.md [hunk 1]
```diff
@@ -152,12 +152,15 @@ Secret -> STREAM_RAILS_SECRET
 # Pusher
 
 1. [Sign up for a free account with this link](https://dashboard.pusher.com/accounts/sign_up).
+
```
[the remark this answers is not in the document]
Comment (maestromac) [reply docs/get-api-keys-dev-env.md]: No need to open a separate PR. This is good 😃

File: package.json [hunk 2]
```diff
@@ -68,6 +69,7 @@
     "eslint-plugin-jsx-a11y": "^6.1.1",
     "eslint-plugin-react": "^7.11.1",
     "faker": "^4.1.0",
+    "gitdocs": "^2.0.0",
```
Comment (maestromac) [inline package.json]: Initially I think that it's not necessary to include gitdocs as a dependency even though the current `@next` version is broken. The `serve-docs` script won me over.

**Comment by maestromac** [comment]
@nholden I can't figure out why but for some reason, having gitdoc as a dependency is failing our deployment. I ran [this build](https://travis-ci.com/thepracticaldev/dev.to/builds/87692292) twice and it fails with the same error. The [subsequent build](https://travis-ci.com/thepracticaldev/dev.to/builds/87696632) with gitdoc removed passed 🤷‍♂️.

**Comment by nholden** [comment]
@maestromac Do we have a way of checking what the `NODE_ENV` environment variable is set to during deployment? If it's `production`, Yarn shouldn't install development dependencies. Alternatively, we could run [`yarn install --production`](https://yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-production-true-false) instead of `yarn install`.

**Comment by maestromac** [comment]
@nholden Thank you for pointing that out! We have assumed the Rails would automatically set `NODE_ENV` based on `RAILS_ENV` but we were wrong. I will add back gitdoc as a devDependencies in the upcoming pr.

**Comment by nholden** [comment]
@maestromac No problem, and thanks for following up!

I'm not sure what dev.to's deployment to Heroku looks like, but I believe if you include the `heroku/nodejs` buildpack before the `heroku/ruby` buildpack, [`NODE_ENV` should automatically be set to `production`](https://devcenter.heroku.com/changelog-items/688). See [Using Multiple Buildpacks for an App](https://devcenter.heroku.com/articles/using-multiple-buildpacks-for-an-app) if you're interested.

**Comment by maestromac** [comment]
Hey @nholden sorry to ping you again. I've [tried to add Gitdoc back](https://github.com/thepracticaldev/dev.to/pull/925) but the CI keep failing. Want to give you a heads up that I might not have it as a dependency and revert the package.json changes.

**Comment by nholden** [comment]
No worries!

That's a bummer that CI isn't working. It looks like Travis is still installing Yarn `devDependencies`, which indicates to me that `NODE_ENV` is still `"development"`. That seems like it would be worth addressing.

If we exhaust all options for including Gitdoc as a dependency, maybe we could consider writing a script outside of Yarn that installs Gitdoc, navigates to the right directory, and runs `gitdocs serve` to save folks a couple steps.

**Pull request merged**